bin/reduce_exposure.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python
+import random
+from openquake.commonlib import nrml
+from openquake.commonlib import sap
+
+
+def random_filter(objects, reduction_factor, seed=42):
+    """
+    Given a list of objects, returns a sublist by extracting randomly
+    some elements. The reduction factor (< 1) tells how small is the extracted
+    list compared to the original list.
+    """
+    assert 0 < reduction_factor <= 1, reduction_factor
+    rnd = random.Random(seed)
+    out = []
+    for obj in objects:
+        if rnd.random() <= reduction_factor:
+            out.append(obj)
+    return out
+
+
+def main(exposure_fname, reduction_factor):
+    """
+    Produce a new exposure from `exposure_fname` by sampling the assets.
+    """
+    exposure = nrml.read(exposure_fname).exposureModel
+    all_assets = len(exposure.assets)
+    exposure.assets.nodes = random_filter(
+        exposure.assets, float(reduction_factor))
+    with open(exposure_fname + '~', 'w') as f:
```

Comment:
A more explanatory output filename could be
exposure_fname + '_reduced' + reduction_factor